ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,383 @@
+    var initialResponse = this.state.initialResponse;
+    var revisedResponse = this.state.revisedResponse;
+    for(var messageIndex = 0; messageIndex < messageList.length; messageIndex++){
+      this.messageId+=1;
+      const message = _.extend({key: this.messageId}, messageList[messageIndex]);
+      if(message.text !== ''){
+        messages.push(message);
+      }
+      if(message.type === 'user'){
+        if(initialResponse === undefined){
+          initialResponse = message.text;
+        }else if(revisedResponse === undefined){
+          revisedResponse = message.text;
+        }
+      }
+    }
+    this.setState({messages, initialResponse, revisedResponse});
+  },
+  
+  addResponse(response){
+    this.addMessages([{type: 'user', text: response}]);
+  },
+  
+  isReadyToMoveOn(){
+    if(this.props.helpType === 'feedback'){
+      if(this.state.revisedResponse !== undefined){ 
+        return true 
+      }
+    }else{
+      if(this.state.initialResponse !== undefined){
+        return true 
+      }
+    }
+    return false
+  },
+  
+  onNextQuestion(){
+    this.props.onQuestionDone(this.state.elapsedMs);
+    this.setState(this.getInitialState());
+  },
+  
+  sendRevision(){
+    this.addMessages([{type: 'info', text: "Here's an example of something you could say/do:"}, {type: 'student', text: _.shuffle(this.props.question.examples)[0]}, {type: 'info', text: 'Would you like to revise your response?'}]);
+  },
+  
+  showExample(){
+    //Maybe we should make a category type message for 'example teacher'. We could also have a different color for it to help differentiate.
+    const example = this.nextExample(true);
+    const messageText = example.type + ' Example:\n' + example.text
+    this.addMessages([{type: 'info', text: messageText}]);
+  },
+  
+  nextExample(increment){
+    const currentIndex = this.state.exampleIndex
+    if(this.state.allExamples.length > currentIndex){
+      if(increment){
+        this.setState({exampleIndex: this.state.exampleIndex + 1});
+      }
+      return this.state.allExamples[currentIndex];
+    }else{
+      return null;
+    }
+  },
+
+  logResponse() {
+    this.logData('message_popup_response');
+  },
+  
+  logRevision(finalText) {
+    this.logData('message_popup_revision', {finalResponseText: finalText});
+  },
+  
+  logRevisionDeclined(){
+    this.logData('message_popup_revision_declined');
+  },
+  
+  logData(type, params = {}) {
+    const {elapsedMs, initialResponse} = this.state;
+    const {question, shouldShowStudentCard, helpType} = this.props;
+    const {finalResponseText} = params;
+    const response:Response = {
+      question,
+      shouldShowStudentCard,
+      helpType,
+      elapsedMs,
+      initialResponse,
+      finalResponseText
+    };
+    this.props.onLog(type, response);
+  },
+
+  resetExperience(){
+    this.setState(this.getInitialState());
+    this.props.resetExperience();
+  },
+
+  getInitialMessages(nextProps){
```

Comment:
I'm not quite following what this method is doing, is it checking the props and then figuring out what initial messages to show based on the question?